buffalo/cmd/info.go

```diff
@@ -30,102 +30,74 @@ var infoCmd = &cobra.Command{
-			c.Stderr = bb
-			c.Run()
-		} else {
-			bb.WriteString("NPM Not Found\n")
-		}
+		return nil
+	},
+}
 
-		bb.WriteString("\n### Yarn Version\n")
-		if _, err := exec.LookPath("yarn"); err == nil {
-			c = exec.Command("yarn", "--version")
-			c.Stdout = bb
-			c.Stderr = bb
-			c.Run()
-		} else {
-			bb.WriteString("Yarn Not Found\n")
-		}
+func checkGoInfo() error {
+	bb := os.Stdout
 
-		bb.WriteString("\n### Dep Version\n")
-		if _, err := exec.LookPath("dep"); err == nil {
-			c = exec.Command("dep", "version")
-			c.Stdout = bb
-			c.Stderr = bb
-			c.Run()
-		} else {
-			bb.WriteString("dep Not Found\n")
-		}
+	bb.WriteString("\n### Go Version\n")
+	c := exec.Command(envy.Get("GO_BIN", "go"), "version")
+	c.Stdout = bb
+	err := c.Run()
+	if err != nil {
+		return errors.WithStack(err)
+	}
 
-		bb.WriteString("\n### Dep Status\n")
-		if _, err := exec.LookPath("dep"); err == nil {
-			c = exec.Command("dep", "status")
-			c.Stdout = bb
-			c.Stderr = bb
-			c.Run()
-		} else {
-			bb.WriteString("dep Not Found\n")
-		}
+	bb.WriteString("\n### Go Env\n")
+	c = exec.Command(envy.Get("GO_BIN", "go"), "env")
+	c.Stdout = bb
+	c.Stderr = bb
+	c.Run()
 
-		bb.WriteString("\n### PostgreSQL Version\n")
-		if _, err := exec.LookPath("pg_ctl"); err == nil {
-			c = exec.Command("pg_ctl", "--version")
-			c.Stdout = bb
-			c.Stderr = bb
-			c.Run()
-		} else {
-			bb.WriteString("PostgreSQL Not Found\n")
-		}
+	return nil
+}
 
-		bb.WriteString("\n### MySQL Version\n")
-		if _, err := exec.LookPath("mysql"); err == nil {
-			c = exec.Command("mysql", "--version")
-			c.Stdout = bb
-			c.Stderr = bb
-			c.Run()
-		} else {
-			bb.WriteString("MySQL Not Found\n")
-		}
+func checkExternalsTools() {
+	bb := os.Stdout
+
+	infoCommands := []struct {
+		Name      string
+		PathName  string
+		Cmd       *exec.Cmd
+		InfoLabel string
+	}{
+		{"Node", "node", exec.Command("node", "--version"), "\n### Node Version\n"},
+		{"NPM", "npm", exec.Command("npm", "--version"), "\n### NPM Version\n"},
+		{"Yarn", "yarn", exec.Command("yarn", "--version"), "\n### Yarn Version\n"},
+		{"dep", "dep", exec.Command("dep", "version"), "\n### Dep Version\n"},
+		{"dep", "dep", exec.Command("dep", "status"), "\n### Dep Status\n"},
+		{"PostgreSQL", "pg_ctl", exec.Command("pg_ctl", "--version"), "\n### PostgreSQL Version\n"},
+		{"MySQL", "mysql", exec.Command("mysql", "--version"), "\n### MySQL Version\n"},
+		{"SQLite", "sqlite3", exec.Command("sqlite3", "--version"), "\n### SQLite Version\n"},
+	}
+
+	for _, cmd := range infoCommands {
+		bb.WriteString(cmd.InfoLabel)
+		execIfExists(cmd.Name, cmd.InfoLabel, cmd.Cmd)
+	}
 
-		bb.WriteString("\n### SQLite Version\n")
-		if _, err := exec.LookPath("sqlite3"); err == nil {
-			c = exec.Command("sqlite3", "--version")
-			c.Stdout = bb
-			c.Stderr = bb
-			c.Run()
-		} else {
-			bb.WriteString("SQLite Not Found\n")
-		}
+}
 
-		return nil
-	},
+func execIfExists(name string, pathName string, c *exec.Cmd) {
```

Comment:
You can clean up this function by passing in the `infoCommand` struct instead of all these args.